### PR TITLE
Split and parametrize test_operators

### PR DESCRIPTION
**Pull request by jbrockmendel**
Orthogonal to #19166 since this does not touch `TestTimedeltaSeriesArithmetic`.

Fix (some) flake8 complaints about 1-letter variable names.

### CODE REVIEW

File: pandas/tests/series/test_operators.py [hunk 1]
```diff
@@ -343,27 +344,26 @@ def test_nat_comparisons_scalar(self):
                  pd.Period('2011-03', freq='M')]]
 
         for l in data:
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: you an parametrize over data here

File: pandas/tests/series/test_operators.py [hunk 2]
```diff
@@ -1300,25 +1290,52 @@ def test_datetime_series_with_timedelta(self):
...
 
+        with pytest.raises(TypeError):
+            pd.offsets.Second(5) - ser
+
+    def test_dt64series_with_DateOffset_smoke(self):
+        # GH 4532
+        # smoke tests for valid DateOffsets
+        ser = Series([Timestamp('20130101 9:01'), Timestamp('20130101 9:02')])
+
+        # valid DateOffsets
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: parameterize (I think you already have these classes as a fixture in offsets somewhere)

File: pandas/tests/series/test_operators.py [hunk 3]
```diff
@@ -1281,7 +1271,7 @@ def test_sub_datetime_compat(self):
         assert_series_equal(s - dt, exp)
         assert_series_equal(s - Timestamp(dt), exp)
 
-    def test_datetime_series_with_timedelta(self):
+    def test_dt64series_with_timedelta(self):
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: don't run words together, dt64 is ok, but generally don't abbreviate

File: pandas/tests/series/test_operators.py [hunk 4]
```diff
@@ -1549,42 +1559,38 @@ def test_invalid_ops(self):
...
-
-        s1 = enddate - startdate
-        s1[2] = np.nan
-
-        for m in [1, 3, 10]:
-            for unit in ['D', 'h', 'm', 's', 'ms', 'us', 'ns']:
+    def test_dt64series_astype_object(self):
+        dt64ser = Series(date_range('20130101', periods=3))
+        result = dt64ser.astype(object)
+        assert isinstance(result.iloc[0], datetime)
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: can you construct the expected directly and compare. astype tests should be in test_dtypes.py

Comment (jreback) [reply pandas/tests/series/test_operators.py]: and I think these are dupes anyhow

File: pandas/tests/series/test_operators.py [hunk 5]
```diff
@@ -1997,25 +2003,39 @@ def test_series_radd_str(self):
...
+        expected = pd.Series([pd.Timedelta('4 days'), pd.Timedelta('5 days'),
+                              pd.Timedelta('6 days')])
+
+        result = pd.Timedelta('3 days') + ser
+        assert_series_equal(result, expected)
+
+        result = ser + pd.Timedelta('3 days')
+        assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('dtype', [None, object])
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: at the very least should split test_operators to timestamp, timedelta, numeric sub-modules (next PR)

File: pandas/tests/series/test_operators.py [hunk 6]
```diff
@@ -2240,3 +2229,51 @@ def test_idxminmax_with_inf(self):
             assert np.isnan(s.idxmin(skipna=False))
             assert s.idxmax() == 0
             np.isnan(s.idxmax(skipna=False))
+
+
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: move to tests/frame

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_operators.py [hunk 7]
```diff
@@ -1613,7 +1602,7 @@ def timedelta64(*args):
 
             try:
                 assert_series_equal(lhs, rhs)
-            except:
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: what the heck is this? don't try/except and then raise an assertion :<

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_operators.py (hunk 7 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: I can only speculate that the original author wanted to customize the exception message.

Comment (jreback) [reply pandas/tests/series/test_operators.py]: pls remove it’s not necessary
the asserts all do this now

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
lgtm. needs a rebase as I just merged a pr fixing some delta things. png on green.

File: pandas/tests/series/test_operators.py [hunk 8]
```diff
@@ -1400,25 +1386,51 @@ def test_datetime_series_with_timedelta(self):
...
         # operate with pd.offsets
-        s = Series([Timestamp('20130101 9:01'), Timestamp('20130101 9:02')])
-
-        result = s + pd.offsets.Second(5)
-        result2 = pd.offsets.Second(5) + s
+        ser = Series([Timestamp('20130101 9:01'), Timestamp('20130101 9:02')])
         expected = Series([Timestamp('20130101 9:01:05'),
                            Timestamp('20130101 9:02:05')])
+
+        result = ser + pd.offsets.Second(5)
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: followup note to cycle thru all Tick offsets

File: pandas/tests/series/test_operators.py [hunk 9]
```diff
@@ -1400,25 +1386,51 @@ def test_datetime_series_with_timedelta(self):
...
 
-        result = s - pd.offsets.Second(5)
-        result2 = -pd.offsets.Second(5) + s
+    def test_dt64_series_sub_tick_DateOffset(self):
+        # GH 4532
+        # operate with pd.offsets
+        ser = Series([Timestamp('20130101 9:01'), Timestamp('20130101 9:02')])
         expected = Series([Timestamp('20130101 9:00:55'),
                            Timestamp('20130101 9:01:55')])
+
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: same here

File: pandas/tests/series/test_operators.py [hunk 10]
```diff
@@ -1400,25 +1386,51 @@ def test_datetime_series_with_timedelta(self):
...
+        result = ser - pd.offsets.Second(5)
         assert_series_equal(result, expected)
+
+        result2 = -pd.offsets.Second(5) + ser
         assert_series_equal(result2, expected)
 
+        with pytest.raises(TypeError):
+            pd.offsets.Second(5) - ser
+
+    @pytest.mark.parametrize('cls_name', ['Day', 'Hour', 'Minute', 'Second',
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: follow up to make this a fixture (to use above as well), prob could move the fixtures in tests/tseries/offsets to the top-level conftest

**Review by jreback — APPROVED** [review]
fine. some follow up notes for later. ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks. next PR should be to split operators tests up into sub-files.

**Pull request merged**